ark/type/keywords/inference.ts

```diff
@@ -434,6 +434,9 @@ export type Default<v = any> = {
 	default?: { value: v }
 }
 
+export type DefaultFor<t> =
+	t extends Primitive ? t | (() => t) : (t & Primitive) | (() => t)
```

Comment:
I need `ifAnyOrUnknown` helper here, do you have one?

Reply:
currently it works fine except allowing string&[] for []

Reply:
I think this is what you want:

```ts

export type DefaultFor<t> =
	| (() => t)
	| (Primitive extends t ? Primitive
	  : t extends Primitive ? t
	  : never)
```

Reply:
I've made 
```ts
export type DefaultFor<t> =
	| (unknown extends t ? Primitive
	  : 0 extends 1 & t ? Primitive
	  : t extends Primitive ? t
	  : never)
	| (() => t)
```
year, yours seems better

Reply:
Pretty good though, I always find your solutions very impressive.